lib/eversign/client.rb

```diff
@@ -1,207 +1,207 @@
-			sub_uri = "/api/download_raw_document?access_key=#{access_key}&business_id=#{business_id}&document_hash=#{document_hash}"
-			download(sub_uri, path)
-		end
-
-		def download_final_document_to_path(document_hash, path, audit_trail=1)
-			sub_uri = "/api/download_raw_document?access_key=#{access_key}&business_id=#{business_id}&document_hash=#{document_hash}&audit_trail=#{audit_trail}"
-			download(sub_uri, path)
-		end
-
-		def upload_file(file_path)
-			payload = { upload: Faraday::UploadIO.new(file_path, 'text/plain') }
-			path = "/api/file?access_key=#{access_key}&business_id=#{business_id}"
-			response = execute_request(:post, path, payload, true)
-			extract_response(response.body, Eversign::Mappings::File)
-		end
-
-		def send_reminder_for_document(document_hash, signer_id)
-			path = "/api/send_reminder?access_key=#{access_key}&business_id=#{business_id}"
-			response = execute_request(:post, path, {document_hash: document_hash, signer_id: signer_id}.to_json)
-			eval(response.body)[:success] ? true : extract_response(response.body)
-		end
-
-		private
-			def execute_request(method, path, body=nil, multipart=false)
-				@faraday ||= Faraday.new(base_uri) do |conn| 
-				  conn.headers = { }
-				  conn.headers['User-Agent'] = 'Eversign_Ruby_SDK'
-				  conn.headers['Authorization'] = token if token
-				  conn.request :multipart if multipart
-				  conn.adapter :net_http
-				end
-
-				@faraday.send(method) do |request|
-					request.url path
-					request.body = body if body
-				end
-			end
-
-			def check_arguments(arguments=[], options={})
-        arguments.each do |argument|
-          raise ('Please specify ' + argument) unless options.has_key?(argument.to_sym) 
+      raise "no success"
+    end
+
+    def get_buisnesses
+      response = execute_request(:get, "/api/business?access_key=#{access_key}")
+      extract_response(response.body, Eversign::Mappings::Business)
+    end
+
+    def get_all_documents
+      get_documents("all")
+    end
+
+    def get_completed_documents
+      get_documents("completed")
+    end
+
+    def get_draft_documents
+      get_documents("draft")
+    end
+
+    def get_cancelled_documents
+      get_documents("cancelled")
+    end
+
+    def get_action_required_documents
+      get_documents("my_action_required")
+    end
+
+    def get_waiting_for_others_documents
+      get_documents("waiting_for_others")
+    end
+
+    def get_templates
+      get_documents("templates")
+    end
+
+    def get_archived_templates
+      get_documents("templates_archived")
+    end
+
+    def get_draft_templates
+      get_documents("template_draft")
+    end
+
+    def get_document(document_hash)
+      path = "/api/document?access_key=#{access_key}&business_id=#{business_id}&document_hash=#{document_hash}"
+      response = execute_request(:get, path)
+      extract_response(response.body, Eversign::Mappings::Document)
+    end
+
+    def create_document(document, isFromTemplate = false)
```

Comment:
differentiate if document is created from template, for later flattening the fields array